src/pcp.rs

```diff
@@ -746,13 +750,13 @@ where
 
     // Check that each of the proof polynomials are well-formed.
     let mut verifier_len = 1;
-    for idx in 0..g.len() {
-        let next_len = 1 + g[idx].arity();
+    for gadget in &mut gadgets {
+        let next_len = 1 + gadget.arity();
```

Comment:
Here again I prefer to use an index `idx`. The reason is that the comments refer to the `idx`-th gadget.